content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+```ts
+public Update(delta: number)
+    {
+        if (this.sprite.y >= GameApp.GroundPosition)
+        {
+            // if downward acceleration brought us to the ground, stop and set airborne to false
+            this.sprite.y = GameApp.GroundPosition;
+            this.verticalSpeed = 0;
+            this.airborne = false;
+        }
+
+        if (this.airborne)
+        {
+            // if we are in the air, accelerate downward by increasing the velocity by a constant value
+            this.verticalSpeed += delta* 1/3;
+        }
+
+        if (GameApp.PressedSpace && !this.airborne)
+        {
+            // jump!
+            this.airborne = true;
+            this.verticalSpeed = -5;
+        }
+
+        // remember the delta update!
+        // the position will change in accordance to how much time passed and the character's speed
+        this.sprite.y += this.verticalSpeed * delta;
+    }
+```
+
+So far, we can create a `Player` object on `GameApp.SetupGame()` and have it update on the `GameApp.Update()` method. The character will stand on the bottom left side of the window and jump when we press Spacebar as it animates slowly. You might be wondering when we are going to make the character move: The answer is we don't! We'll move obstacles towards the character.
+
+## Obstacles
+
+We have a character that jumps, so a good next point to explore is obstacles that scroll horizontally to the left. We want to have an arbitrary number of obstacles coming at the player, leaving some space for it to react accordingly.
+
+Since we are doing scrolling obstacles, we can also make our code so that it includes aesthetic objects such as clouds. We can differentiate those with objects that the character can bump into by having a boolean member called `solid`. We can even make them scroll a little bit slower so that we have some sort of [parallax scrolling](https://en.wikipedia.org/wiki/Parallax_scrolling). For this, let's define a base scroll speed on `GameApp` and subtract one from it if the world object is not solid.
+Lastly, let's add a modifier to the speed of the obstacles so that they move a little bit faster as the score increases:
+
+```ts
+
+export class GameApp {
+    static ScrollSpeed : number = 3;
+    ...
+}
+
+class ScrollingObject {
+    sprite: PIXI.Sprite;
+    solid = true;
+
+    public constructor(spriteName: string, isSolid: boolean) {
+        // todo explain
+        this.sprite =  GetSprite(spriteName);
+        this.sprite.y = GameApp.Width;
+        this.sprite.x = GameApp.GroundPosition;
+
+        this.sprite.anchor.set(0, 1);
+        this.solid = isSolid;
+    }
+
+    public Update(delta:number) {
+        var baseScrollSpeed = (this.solid) ? GameApp.ScrollSpeed : GameApp.ScrollSpeed-1;
+
+        // modifier for speed depending on score so that it gets more difficult
+        var scrollSpeed = baseScrollSpeed + Math.min(GameApp.Score/15.0 , 1);
```

Comment:
maybe use `const` or `let`? `var` is rarely used in the modern day, and even less when you have the typescript compiler already included

Reply:
One way of actually dealing with all these formatting matters is by actually running `prettier` on your entire example directory. You can check how we do it in this repo and simply either install prettier globally or locally and run it once off. Most of the longer non-comment lines will be dealt by automatically, and some spaces and such will also be solved automatically.